dotCMS/src/main/java/com/dotcms/publishing/job/SiteSearchJobImpl.java

```diff
@@ -208,24 +285,93 @@ else if(bs.getBundlerClass().equals(HTMLPageAsContentBundler.class.getName()))
+            audit.setPath(paths.size() > 0 ? UtilMethods.join(paths,",") : "/*");
             audit.setPathInclude(include);
-            APILocator.getSiteSearchAuditAPI().save(audit);
+            siteSearchAuditAPI.save(audit);
         }
         catch(DotDataException ex) {
-            Logger.warn(this, "can't save audit data",ex);
+            Logger.error(this, "can't save audit data",ex);
         }
         finally {
             HibernateUtil.closeSession();
         }
 
         date = DateUtil.getCurrentDate();
-        ActivityLogger.logInfo(getClass(), "Job Finished", "User: " +APILocator.getUserAPI().getSystemUser().getUserId()+ "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME  );
-        AdminLogger.log(getClass(), "Job Finished", "User: " +APILocator.getUserAPI().getSystemUser().getUserId()+ "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME );
+        ActivityLogger.logInfo(getClass(), "Job Finished", "User: " +userAPI.getSystemUser().getUserId()+ "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME  );
+        AdminLogger.log(getClass(), "Job Finished", "User: " +userAPI.getSystemUser().getUserId()+ "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME );
+    }
+
+     private String newIndexName(){
+        return SiteSearchAPI.ES_SITE_SEARCH_NAME + StringPool.UNDERLINE
+                + ESMappingAPIImpl.datetimeFormat.format(new Date());
+     }
+
+     private IndexMetaData getIndexMetaData(String indexAlias) throws DotDataException {
+        String indexName = null;
+        boolean defaultIndex = false;
+        long recordCount = 0;
+        if (UtilMethods.isSet(indexAlias)) {
+            indexAlias = indexAlias.split("\\s+")[0];
+            final List<String> indices = siteSearchAPI.listIndices();
+            final Map<String, String> aliasMap = esIndexAPI.getAliasToIndexMap(indices);
+            indexName = aliasMap.get(indexAlias);
+            if (UtilMethods.isSet(indexName)) {
+                if (siteSearchAPI.isDefaultIndex(indexAlias)) {
+                    Logger.info(this, "Index is current Site-Search DEFAULT.");
+                    defaultIndex = true;
+                }
+            } else {
+                // the alias comes with an index name that is already in use.
+                if(indices.contains(indexAlias)){
+                   indexName = indexAlias;
+                   indexAlias = null;
+                }
+            }
+            if(UtilMethods.isSet(indexName)){
+                final SiteSearchResults search = siteSearchAPI.search(indexName, "*",0, 10);
+                recordCount = search.getTotalResults();
+            }
+        }//if indexName is null. Then the result is interpreted as a new index.
+        return new IndexMetaData(indexName, defaultIndex, indexAlias, recordCount == 0);
+    }
+
+    static class IndexMetaData {
+
+        private String indexName;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Private field 'indexName' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093862)